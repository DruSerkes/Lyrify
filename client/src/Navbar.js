import React from 'react';
import { NavLink } from 'react-router-dom';
import { useSelector, shallowEqual } from 'react-redux';
import { AppBar, Avatar, Button, Toolbar, Grid, Typography, Box } from '@material-ui/core';

const Navbar = () => {
	const user = useSelector((state) => state.user.data, shallowEqual);
	return (
		<AppBar position="sticky">
			<Toolbar>
				<Box display="flex" className="Navbar-Logo">
					<Avatar src="logo500.png" alt="Lyrify" />
					<Typography variant="body1"> yrify</Typography>
				</Box>
				{user !== undefined ? (
					<React.Fragment>
						<Grid container>
							<Button
								variant="text"
								size="large"
								className="Navbar-Btn"
								color="inherit"
								component={NavLink}
								to="/playing"
							>
								Now Playing
							</Button>
							<Button
								variant="text"
								size="large"
								className="Navbar-Btn"
								color="inherit"
								component={NavLink}
								to="/search"
							>
								Search
							</Button>
						</Grid>
						<Avatar sizes="lg" src={user.img_url} alt={user.display_name} />
					</React.Fragment>
				) : null}
			</Toolbar>
		</AppBar>
	);
};

export default Navbar;
