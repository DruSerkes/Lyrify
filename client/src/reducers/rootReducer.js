import songReducer from './songReducer';
import userReducer from './userReducer';
import favoritesReducer from './favoritesReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ song: songReducer, user: userReducer, favorites: favoritesReducer });

export default rootReducer;
