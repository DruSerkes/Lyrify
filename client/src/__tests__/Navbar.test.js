import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../Navbar';
import { MemoryRouter } from 'react-router-dom';
import rootReducer from '../reducers/rootReducer';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));

it('renders without crashing', () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<Navbar />
			</MemoryRouter>
		</Provider>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<Navbar />
			</MemoryRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
