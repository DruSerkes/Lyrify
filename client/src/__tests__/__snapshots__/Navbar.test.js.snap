// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        aria-current="page"
        class="MuiBox-root MuiBox-root-2 Navbar-Logo active"
        href="/"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            alt="Lyrify"
            class="MuiAvatar-img"
            src="logo500.png"
          />
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
           yrify
        </p>
      </a>
    </div>
  </header>
</DocumentFragment>
`;
